bin/tweet_trigger

```diff
@@ -0,0 +1,444 @@
+    
+    parser.add_argument("--preferred-longitude", dest="preferredLongitude",
+                      help="""The preferred longitude for this event.""", metavar="PREFERREDLON")
+    
+    parser.add_argument("--preferred-depth", dest="preferredDepth",
+                      help="""The preferred depth for this event.""", metavar="PREFERREDDEPTH")
+    
+    parser.add_argument("--preferred-eventtime", dest="preferredEventTime",
+                      help="""The preferred event time (2010-01-14T14:11:28.691Z).""", metavar="PREFERREDDEPTH")
+    
+
+
+    # Filter out any options that are not in the above list
+    # Get the list of defined options
+    options = {}
+    validargs, invalidargs = parser.parse_known_args()
+    options = vars(validargs)
+    
+    # Print help menu and exit if none of options are defined
+    foundOne = False
+    for argvalue in options.values():
+        if argvalue != None:
+             foundOne = True
+             break
+    if not foundOne:
+        parser.print_help()
+        sys.exit(0)
+
+    # Read in config file
+    homedir = os.path.dirname(os.path.abspath(__file__))
+    configfile = os.path.join(homedir, 'configTweeter.ini')
+    if not os.path.isfile(configfile):
+        print('File "%s" does not exist.  Exiting' % configfile)
+        sys.exit(1)
+
+    config = configparser.ConfigParser()
+    config.readfp(open(configfile))
+
+    if not config.has_section('SETUP'):
+        print('Config file "%s" is missing section "SETUP".  Exiting' % configfile)
+        sys.exit(1)
+
+    reqoptions = ['magthresh', 'timethresh', 'tooclosedistance', 'toooldtime', 
+                  'ignoreregions', 'urltemplate', 'logging_on', 'logfile']
+    missing = []
+    for option in reqoptions:
+        if not config.has_option('SETUP', option):
+            missing.append(option)
+    if len(missing):
+        print('Config file "%s" is missing SETUP options "%s".  Exiting' % (configfile, ','.join(missing)))
+        sys.exit(1)
+
+    ignoreregions = config.get('SETUP', 'ignoreregions').split(',')
+    loggingOn = bool(int(config.get('SETUP', 'logging_on')))
+    logfile = os.path.join(homedir, config.get('SETUP', 'logfile'))
+
+    # Set up log file
+    if loggingOn:
+        f = open(logfile, 'at')
+
+    # Connect to database
+    prefix = 'testdb'
+    port = config.get('DATABASE',prefix+'_port')
+    user = config.get('DATABASE',prefix+'_user')
+    dbname = config.get('DATABASE',prefix+'_name')
+    password = config.get('DATABASE',prefix+'_password')
+
+    conn = psycopg2.connect(dbname=dbname, user=user, port=port, password=password)
+    conn.autocommit = True
+    cur = conn.cursor()
+    
+    # Variables used later on
+    tnow = datetime.datetime.now()
+    utcnow = datetime.datetime.utcnow()
+    eventdatetime = datetime.datetime.strptime(options['preferredEventTime'], '%Y-%m-%dT%H:%M:%S.%fZ')
+
+    main(options)    
```

Comment:
Only do sysexit from the actual main, no functions should do sysexit, they should return errors, log errors etc. and let the caller decide what to do. 
then then in the main where you do DB connections, you can do the close and it happens in only one spot then easier to trace your steps